ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/answers/activity-matrix-answer/activity-matrix-answer.component.spec.ts

```diff
@@ -0,0 +1,727 @@
+import { QuestionPromptComponent } from '../question-prompt/questionPrompt.component';
+import { ActivityMatrixAnswer } from './activity-matrix-answer.component';
+
+const mockQuestions: Question[] = [
+  {
+    stableId: 'Q1',
+    questionLabel: 'How often do you eat fruits?',
+    tooltip: null,
+  },
+  {
+    stableId: 'Q2',
+    questionLabel: 'How often do you eat vegetables?',
+    tooltip: null,
+  },
+];
+const mockGroups: Group[] = [
+  {
+    identifier: 'G1',
+    name: 'Group 1',
+  },
+  {
+    identifier: 'G2',
+    name: 'Group 2',
+  },
+];
+const mockOptions: Option[] = [
+  {
+    exclusive: false,
+    stableId: 'ONCE_WEEK',
+    optionLabel: 'Once a week',
+    tooltip: null,
+    groupId: mockGroups[0].identifier,
+  },
+  {
+    exclusive: false,
+    stableId: 'TWICE_WEEK',
+    optionLabel: 'Twice a week',
+    tooltip: null,
+    groupId: mockGroups[1].identifier,
+  },
+];
+const mockExclusiveOptions: Option[] = [
+  {
+    exclusive: true,
+    stableId: 'O1',
+    tooltip: null,
+    groupId: mockGroups[0].identifier,
+    optionLabel: 'I am exclusive!',
+  },
+  {
+    exclusive: false,
+    stableId: 'O2',
+    tooltip: null,
+    groupId: mockGroups[0].identifier,
+    optionLabel: 'I am not exclusive!',
+  },
+  {
+    exclusive: false,
+    stableId: 'O3',
+    tooltip: null,
+    groupId: mockGroups[0].identifier,
+    optionLabel: 'I am not exclusive as well!',
+  },
+  {
+    exclusive: false,
+    stableId: 'O4',
+    tooltip: null,
+    groupId: mockGroups[1].identifier,
+    optionLabel: 'I am from different group...',
+  },
+];
+
+describe('ActivityMatrixAnswer', () => {
+  let fixture: ComponentFixture<ActivityMatrixAnswer>;
+  let block: ActivityMatrixQuestionBlock;
+  let component: ActivityMatrixAnswer;
+
+  beforeEach(
+    waitForAsync(() => {
+      TestBed.configureTestingModule({
+        declarations: [QuestionPromptComponent, ActivityMatrixAnswer],
+        imports: [MatRadioModule, MatCheckboxModule],
+      }).compileComponents();
+    }),
+  );
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(ActivityMatrixAnswer);
+
+    expect(fixture instanceof ComponentFixture).toBeTrue();
+    expect(fixture.componentInstance instanceof ActivityMatrixAnswer).toBeTrue();
+
+    block = new ActivityMatrixQuestionBlock();
+    block.selectMode = SelectMode.Single;
+    block.question = 'Answer following questions';
+    block.stableId = 'MATRIX_BLOCK';
+    block.questions = [];
+    block.groups = [];
+    block.options = [];
+
+    component = fixture.componentInstance;
+    component.readonly = false;
+    component.block = block;
+  });
+
+  it('renders question prompt is one is provided', () => {
+    block.question = 'How are you?';
+
+    component.ngOnChanges({ block: new SimpleChange(null, block, true) });
+    fixture.detectChanges();
+
+    const questionPrompt = fixture.debugElement.query(By.css('.ddp-question-prompt'));
+
+    expect(questionPrompt).not.toBe(null);
```

Comment:
suggest checking explicitly 
` expect(questionPrompt.nativeElement.textContent).toContain('How are you?');`